public/app/core/components/Select/DashboardPicker.tsx

```diff
@@ -56,19 +57,37 @@ export const DashboardPicker = ({
     (async () => {
       // value was manually changed from outside or we are rendering for the first time.
       // We need to fetch dashboard information.
-      const res = await getDashboardAPI().getDashboardDTO(value, undefined);
-      const formatted = ResponseTransformers.ensureV1Response(res);
+      // when using getDashboardAPI, if isV2Mode is not set, we will always return the v1 schema version
+      // isV2Mode is set to true when the feature toggle of dashboardSchemaV2 and useV2DashboardsAPI is enabled
+      // We could create a different feature toggle with better naming if we want, but the idea of using
+      // dashboardSchemaV2 is to indicate that this component is ready to process the new schema
+      const isV2Mode = config.featureToggles.dashboardSchemaV2 && config.featureToggles.useV2DashboardsAPI;
```

Comment:
this could be a helper function really.

Reply:
if we agree we only need a single toggle for now (`useV2DashboardsAPI`), then no helper needed.